### PR TITLE
Don’t eat closing braces while trying to recover to a declaration start keyword

### PR DESCRIPTION
**Pull request by ahoppen**
Previously, when given 

```swift
class A {
  ^
}
class B {
}
```

recovery to the second `class` (because `^` is unexpected) would also consume the closing `}` nesting `B` inside `A`. That’s unexpected. Recovery of decl keywords inside member decl lists should not consume closing braces.

---

While doing this, also fix a few miscellaneous bugs in the test infrastructure.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/_SwiftSyntaxTestSupport/SyntaxComparison.swift [hunk 1]
```diff
@@ -76,14 +76,17 @@ extension TreeDifference: CustomDebugStringConvertible {
                 """
     }
 
+    let baselineRootConverter = SourceLocationConverter(file: "Baseline.swift", source: baseline.root.description)
+    let nodeRootConverter = SourceLocationConverter(file: "Actual.swift", source: node.root.description)
```
Comment (bnbarham) [inline Sources/_SwiftSyntaxTestSupport/SyntaxComparison.swift:80]: May as well use these when printing the nodes above as well?

File: Tests/SwiftParserTest/ParserTests.swift [hunk 2]
```diff
@@ -160,6 +160,8 @@ public class ParserTests: XCTestCase {
         || fileURL.absoluteString.contains("26773-swift-diagnosticengine-diagnose.swift")
         || fileURL.absoluteString.contains("parser-cutoff.swift")
         || fileURL.absoluteString.contains("26233-swift-iterabledeclcontext-getmembers.swift")
+        || fileURL.absoluteString.contains("26190-swift-iterabledeclcontext-getmembers.swift")
+        || fileURL.absoluteString.contains("26139-swift-abstractclosureexpr-setparams.swift")
```
Comment (bnbarham) [inline Tests/SwiftParserTest/ParserTests.swift:164]: These aren't even *that* bad, pretty surprising to me that they cause a stack overflow. Isn't even 100 nested braces!

File: Sources/_SwiftSyntaxTestSupport/SyntaxComparison.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/_SwiftSyntaxTestSupport/SyntaxComparison.swift:80]: Good idea 👍

File: Tests/SwiftParserTest/ParserTests.swift (hunk 2 above)
Comment (ahoppen) [reply Tests/SwiftParserTest/ParserTests.swift:164]: The issue here appears to be debug builds. In release builds, these are just fine.

**Comment by ahoppen** [comment]
@swift-ci please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Pull request merged**